src/cheevos/CheevosFrontendBridge.cpp

```diff
@@ -0,0 +1,110 @@
+/*
+ *  Copyright (C) 2020 Team Kodi (https://kodi.tv)
+ *
+ *  SPDX-License-Identifier: GPL-2.0-or-later
+ *  See LICENSE.md for more information.
+ */
+
+#include "CheevosFrontendBridge.h"
+
+#include <kodi/Filesystem.h>
+#include <stdio.h>
+
+using namespace LIBRETRO;
+
+void *CCheevosFrontendBridge::OpenFile(const char* path_utf8)
+{
+  // Return NULL for error
+  if (path_utf8 == nullptr)
+    return nullptr;
+
+  // TODO: Handle UTF-8?
+  std::unique_ptr<FileHandle> fileHandle(new FileHandle{ path_utf8 });
+  fileHandle->file.reset(new kodi::vfs::CFile);
+
+  if (!fileHandle->file->OpenFile(fileHandle->path, 0))
+    return nullptr;
+
+  // Return the opaque file handle on success
+  return static_cast<void*>(fileHandle.release());
+}
+
+void CCheevosFrontendBridge::CloseFile(void* file_handle)
+{
+  if (file_handle == nullptr)
+    return;
+
+  FileHandle *fileHandle = static_cast<FileHandle*>(file_handle);
+
+  fileHandle->file->Close();
+  delete fileHandle;
+}
+
+size_t CCheevosFrontendBridge::GetPosition(void* file_handle)
+{
+  // Return 0 for error
+  if (file_handle == nullptr)
+    return 0;
+
+  FileHandle *fileHandle = static_cast<FileHandle*>(file_handle);
+
+  const int64_t currentPosition = fileHandle->file->GetPosition();
+
+  if (currentPosition < 0)
+    return 0;
+
+  // Return the current read / write position for the file
+  return currentPosition;
+}
+
+void CCheevosFrontendBridge::Seek(void* file_handle, size_t offset, int origin)
+{
+  if (file_handle == nullptr)
+    return;
+
+  FileHandle *fileHandle = reinterpret_cast<FileHandle*>(file_handle);
```

Comment:
@NikosSiak I accidentally left two `reinterpret_cast<>`s from https://github.com/kodi-game/game.libretro/pull/58 that should have also been converted to `static_cast<>`. Can you change?

The reason for the change is because it's safe to convert `void*` to any other pointer type, but libretro uses a declared-but-undefined struct that the compiler doesn't deduce to `void*`, so we need to break out the big guns (`reinterpret_cast<>`) to brute-force the compiler into accepting the conversion. We pay for the type-safety of libretro with a less-safe cast in https://github.com/kodi-game/game.libretro/pull/58, which is OK.